src/filechooser.cpp

```diff
@@ -204,14 +204,14 @@ namespace LXQt
         ExtractFilters(options, nameFilters, mimeTypeFilters, allFilters, selectedMimeTypeFilter);
 
         // for handling of options - choices
-        QScopedPointer<QWidget> optionsWidget;
+        QWidget *optionsWidget = nullptr;
```

Comment:
Would you mind to still use the guarded pointer (with a paranoia of leak in mind) and releasing the ownership, when adding to the parent widget e.g.:
```diff
diff --git a/src/filechooser.cpp b/src/filechooser.cpp
index 56554ff..c38ff8b 100644
--- a/src/filechooser.cpp
+++ b/src/filechooser.cpp
@@ -204,7 +204,7 @@ namespace LXQt
         ExtractFilters(options, nameFilters, mimeTypeFilters, allFilters, selectedMimeTypeFilter);
 
         // for handling of options - choices
-        QScopedPointer<QWidget> optionsWidget;
+        std::unique_ptr<QWidget> optionsWidget;
         // to store IDs for choices along with corresponding comboboxes/checkboxes
         QMap<QString, QCheckBox *> checkboxes;
         QMap<QString, QComboBox *> comboboxes;
@@ -235,9 +235,11 @@ namespace LXQt
             fileDialog->setNameFilters(nameFilters);
         }
 
+        bool has_options = false;
         if (optionsWidget) {
             if (auto layout = fileDialog->dialog().layout()) {
-                layout->addWidget(optionsWidget.get());
+                layout->addWidget(optionsWidget.release());
+                has_options = true;
             }
         }
 
@@ -255,7 +257,7 @@ namespace LXQt
             results.insert(QStringLiteral("uris"), files);
             results.insert(QStringLiteral("writable"), true);
 
-            if (optionsWidget) {
+            if (has_options) {
                 QVariant choices = EvaluateSelectedChoices(checkboxes, comboboxes);
                 results.insert(QStringLiteral("choices"), choices);
             }
@@ -325,7 +327,7 @@ namespace LXQt
         ExtractFilters(options, nameFilters, mimeTypeFilters, allFilters, selectedMimeTypeFilter);
 
         // for handling of options - choices
-        QScopedPointer<QWidget> optionsWidget;
+        std::unique_ptr<QWidget> optionsWidget;
         // to store IDs for choices along with corresponding comboboxes/checkboxes
         QMap<QString, QCheckBox *> checkboxes;
         QMap<QString, QComboBox *> comboboxes;
@@ -376,9 +378,11 @@ namespace LXQt
             fileDialog->setNameFilters(nameFilters);
         }
 
+        bool has_options = false;
         if (optionsWidget) {
             if (auto layout = fileDialog->dialog().layout()) {
                 layout->addWidget(optionsWidget.get());
+                has_options = true;
             }
         }
 
@@ -390,7 +394,7 @@ namespace LXQt
             }
             results.insert(QStringLiteral("uris"), files);
 
-            if (optionsWidget) {
+            if (has_options) {
                 QVariant choices = EvaluateSelectedChoices(checkboxes, comboboxes);
                 results.insert(QStringLiteral("choices"), choices);
             }
```
Note: using std:unique_ptr as the QScopedPointer::take() was deprecated in Qt v6.1

Reply:
Absolutely. Thanks for the suggestion.